composer.json

```diff
@@ -86,7 +86,7 @@
         ],
         "post-autoload-dump": [
             "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
-            "@php artisan package:discover"
+            "#@php artisan package:discover"
```

Comment:
I don't think we can comment out this line. It is required by laravel to know where to find plugins. Removing it will mean adding new packages won't be auto-loaded by Laravel, as far as I know.

In the latest versions of laravel, they have an `--ansi` flag, which might help running it in docker?
https://github.com/laravel/laravel/blob/master/composer.json

Reply:
Apparently if you disable the db completely, it will ignore that part. https://www.reddit.com/r/laravel/comments/dcc49y/laravel_without_database_connection/

Reply:
From what I've seen is this is not a docker issue. `php artisan` needs a database, otherwise it raises a `connection refused`. I had the same problem with a build on my local machine. Without a runnning database, this step always fails with a connection error.
After some digging, here what I understand (i know nothing about Laravel, sorry my nood understanding):
- `composer install` : composer fetches all packages liste in composer.json
- `php artisan package:discover`: helps laravel to discover any new supported libraries or packages when running `composer update` (source: https://stackoverflow.com/questions/54648518/what-the-meaning-of-discovered-package)
- `php artisan` commands always require a running database
- post-autoload-dump: the script is run during install and update.

So my understanding is that running `php artisan package:discover` in `post-autoload-dump` is not ok because it runs during `install`, while it is only useful for `update`. We can use the script `pre-update-cmd` to do that.

Ok with that ?

Reply:
Hey, I had not seen your second comment. You put me on the right track. Changing the connection to an in-memory sqlite during build seems to do the trick. I will update the PR soon.